modules/apps/27-interchain-accounts/controller/ibc_middleware.go

```diff
@@ -261,10 +261,5 @@ func (im IBCMiddleware) GetAppVersion(ctx sdk.Context, portID, channelID string)
 // into an InterchainAccountPacketData. This function implements the optional
 // PacketDataUnmarshaler interface required for ADR 008 support.
 func (IBCMiddleware) UnmarshalPacketData(bz []byte) (interface{}, error) {
-	var packetData icatypes.InterchainAccountPacketData
-	if err := icatypes.ModuleCdc.UnmarshalJSON(bz, &packetData); err != nil {
-		return nil, err
-	}
-
-	return packetData, nil
+	return icatypes.PacketDataFromBytes(bz)
```

Comment:
We could also make `UnmarshalJSON` a method of `InterchainAccountPacketData`. 

```go
func (iapd InterchainAccountPacketData) UnmarshalJSON(bz []byte) (InterchainAccountPacketData, error)
```

Usage:

```go
var packetData icatypes.InterchainAccountPacketData
if err := packetData.UnmarshalJSON(bz); err != nil { 
   // handle err
}
```

Happy to defer to you guys on what API you prefer

Reply:
Yes, this is also a valid approach although the function signature likely should be different. I tried this first (and called it `FromBytes`) but I thought the one I added looked better in the end. I'm happy to keep this PR open until we get some other opinions

```go
func (iapd *InterchainAccountPacketData) FromBytes(bz []byte) error
```

Reply:
> although the function signature likely should be different

Aha, yes, sorry for my typo! Indeed, you don't need to return `(InterchainAccountPacketData, error)` since you're acting on unmarshaling into the receiver. I'll edit my first comment! 😛 

I think `UnmarshalJSON` naming is pretty common/standard in golang and in cosmos. Here's an example in [`x/staking`](https://github.com/cosmos/cosmos-sdk/blob/60198f077af73867594d8dbe0359e5f84b8cba2d/x/staking/types/delegation.go#L312) and [`x/auth`](https://github.com/cosmos/cosmos-sdk/blob/60198f077af73867594d8dbe0359e5f84b8cba2d/x/auth/types/account.go#L259)

Reply:
Ok, sounds good. I'll do this

Reply:
Note: I did this